llama-index-integrations/vector_stores/llama-index-vector-stores-milvus/llama_index/vector_stores/milvus/base.py

```diff
@@ -436,6 +435,45 @@ def clear(self) -> None:
         """Clears db."""
         self._milvusclient.drop_collection(self.collection_name)
 
+    def get_nodes(
+        self,
+        node_ids: Optional[List[str]] = None,
+        filters: Optional[MetadataFilters] = None,
+    ) -> list[BaseNode]:
+        """Get nodes by node ids or metadata filters.
+
+        Args:
+            node_ids (Optional[List[str]], optional): IDs of nodes to retrieve. Defaults to None.
+            filters (Optional[MetadataFilters], optional): Metadata filters. Defaults to None.
+
+        Raises:
+            ValueError: Neither or both of node_ids and filters are provided.
+
+        Returns:
+            List[BaseNode]:
+        """
+        if node_ids is None and filters is None:
+            raise ValueError("Either node_ids or filters must be provided.")
+
+        filters_cpy = deepcopy(filters) or MetadataFilters(filters=[])
+        milvus_filter = _to_milvus_filter(filters_cpy)
+
+        if node_ids is not None and milvus_filter:
+            raise ValueError("Only one of node_ids or filters can be provided.")
+
+        res = self.client.query(
+            ids=node_ids, collection_name=self.collection_name, filter=milvus_filter
+        )
+        return [
+            metadata_dict_to_node(
+                {
+                    "_node_content": item.get("_node_content", None),
+                    "_node_type": item.get("_node_type", None),
+                }
```

Comment:
curious why not just pass the entire `item` dict?

Reply:
This is indeed unnecessary. Additionally, I've modified a bit referring to the `query` method. Now it supports `text_key` and returned nodes include embeddings.